ChatGPT Blog Post

907Resident opened the pull request:
I added the blog post I wrote for the team regarding ChatGPT. 

I noticed you (@mcnewcp) used a `{{site.baseurl}}` convention to ensure your images rendered in your housing market blog post. I was not sure what does; I noticed that file was ignored with your `.gitignore` so I could not investigated it further. I ended up using a relative path scheme to make my images visible in my post. But let me know if a different convention is required.

You will find my post in the `_posts` folder along with the images in the `assets` folder. Also, in case you are interested, I also left a complete version of the post in `_drafts`. 

> NOTE: I did not do any `jekyll` or Ruby stuff, but I did install it and it is ready. Maybe we can talk about that at some point

907Resident committed:
updated the 'about'  file as @mcnewcp requested

the changes are minor but provide a foundation to what may become a more involved page.

907Resident committed:
added .keep file

the .keep file maintains the directory structure when a directory is empty. the file itself is meaningless other than a placeholder

907Resident committed:
initial commit

this commit contains a completed draft of the chatgpt post

907Resident committed:
initial commit

this commit reflects a transfer of the chatgpt draft into posts. the draft was copied and modified slightly so that the images could be viewable.

in other words, the images are now located in assets/2022-12-01-chatgpt and the '_posts' version of the chatgpt article was modified so that the paths to images were compatible

a source of possible issue, @mcnewcp used a  convetion in front of the 'assets' part of the file path. @907Resident is unsure of what that means but still expects the images to show up properly.  is a directory hidden and only @mcnewcp can access it

907Resident committed:
initial commit

copied images from drafts to assets -- changed the name of the directory containing the images to match previously established naming convention

see de4fbfc for more details on this move and other related details

mcnewcp commented:
@907Resident I'm addressing each of your questions below and I'll make some additional commits to fix issues before merging.  I'll discuss each of these with you in more detail later.  There are a few conventions that need to be followed so that gh-pages can deploy correctly.  It's minor stuff that I've run into by trial and error mostly.

> I noticed you (@mcnewcp) used a {{site.baseurl}} convention to ensure your images rendered in your housing market blog post. I was not sure what does; I noticed that file was ignored with your .gitignore so I could not investigated it further. I ended up using a relative path scheme to make my images visible in my post. But let me know if a different convention is required.

The `{{site.baseurl}}` is referencing a global variable `site` and pulling the `baseurl` from it.  The reason this is required is that the way github pages builds the site on deployment, the assets are organized slightly differently than they are locally, so the only way (I've found) to make the paths work both locally and after deployment is to reference in that way.  When the site is built, the assets are placed in a different location, so the relative file paths you've used need to be changed.

> You will find my post in the _posts folder along with the images in the assets folder. Also, in case you are interested, I also left a complete version of the post in _drafts.

The drafts folder actually has a specific purpose.  It's for storing posts that are complete but not yet finalized.  Then when you build the site you can choose to include drafts or not, for testing purposes.  You'll notice I've also saved an example post in there.  Because of this we probably shouldn't store anything in there that doesn't meet that criteria, otherwise it'll break when we try to build the site.  I'll remove.

> NOTE: I did not do any jekyll or Ruby stuff, but I did install it and it is ready. Maybe we can talk about that at some point

You could contribute posts and pages without ever needing jekyll or ruby.  The only time you'd need to use either is if you want to build the site locally to preview before commit/push.

mcnewcp committed:
reformat filename

@907Resident jekyll uses the filename to categorize blog posts so they should always be date-title, similar to what I've done here.

mcnewcp committed:
adding frontmatter

@907Resident jekyll uses yaml frontmatter to define some important metadata about each post

mcnewcp committed:
remove redundant title

mcnewcp committed:
rename asset directory

@907Resident a convention I've seen repeated is to have all post related assets stored in a directory that matches post-title_files.  I'm not sure how important it is but I've been following it.

mcnewcp committed:
fix md relative paths

@907Resident we'll have to continue this covention for asset pathing for as long as we build the site using gh-pages

mcnewcp committed:
convert some md to html

For some reason jekyll really didn't like the markdown within html tags, so I converted it all to html

mcnewcp committed:
remove unneeded files from _drafts

mcnewcp committed:
remove unneeded file

mcnewcp reviewed (approved):
See details on my changes in above commits if you want more info.  Post looks good!

907Resident commented:
Okay, I finally had a second to look at your response here. 

Thanks for making all of those changes. I did not mean to leave you with those but it looked like you handled it pretty easily. 

I will look at this PR again when I make another addition to the site.
